analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -76,9 +76,11 @@ protected void ReadParameters(SonarAnalysisContext context, CompilationAnalysisC
     public abstract class UtilityAnalyzerBase<TMessage> : UtilityAnalyzerBase
         where TMessage : IMessage, new()
     {
+        private const int TokenCountThreshold = 10_000;
```

Comment:
At some point, the implementation was counting only identifiers. Does the measurement comment in this PR reflects `identifiers` or `all tokens`?

Because 7 000 identifiers could be something like 35 000 tokens...?

Reply:
For example, `analyzers\tests\SonarAnalyzer.UnitTest\ControlFlowGraph\ControlFlowGraphTest.cs` has 34 000 tokens (12 000 identifiers)

I think we should increate the number to something like 40 000-50 000 tokens.

Reply:
> At some point, the implementation was counting only identifiers. Does the measurement comment in this PR reflects identifiers or all tokens?

As mentioned in comment the numbers are for tokens.

Reply:
one thing to consider: are we introducing a performance regression for all codebases just to avoid the edge case of huge files? or is the overall change negligible for normal codebases?

Reply:
actually, never mind that : on some codebases anyway this is already taking 20-30% of analysis time. So what we'll do here should decrease it.